src/hooks/useFetch.ts

```diff
@@ -5,25 +5,32 @@ const useFetch = (url: string, options: object = {}) => {
   const [response, setResponse] = useState<any>({});
   const [error, setError] = useState<any>(null);
   const [isLoading, setIsLoading] = useState(true);
-
+  let cancel: () => void;
   useEffect(() => {
     (async () => {
       setIsLoading(true);
       try {
-        const res = await fetch({
+        const { requestPromise: res, cancelApi } = fetch({
```

Comment:
Any reason for renaming the `requestPromise`?

Reply:
Oops yeah I should not be doing this now.
I'll remove res and use requestPromise only.